core/unit_test/incremental/Test14_MDRangeReduce.hpp

```diff
@@ -38,6 +38,13 @@ struct MyComplex {
   KOKKOS_INLINE_FUNCTION
   MyComplex(const MyComplex& src) : _re(src._re), _im(src._im) {}
 
+  KOKKOS_INLINE_FUNCTION
+  MyComplex& operator=(const MyComplex& src) {
+    _re = src._re;
+    _im = src._im;
+    return *this;
+  }
```

Comment:
What compiler emitted a warning for this?
Same as for `floating_point_wrapper` and `long_wrappper`, I don't fully get why this is related to `-Wdeprecated-copy`
https://godbolt.org/z/av9hbYjah

Reply:
https://godbolt.org/z/beqo1rqoG shows
```
<source>:11:12: error: implicitly-declared 'constexpr Foo& Foo::operator=(const Foo&)' is deprecated [-Werror=deprecated-copy]
   11 |     foo1 = foo2;
      |            ^~~~
<source>:3:5: note: because 'Foo' has user-provided 'Foo::Foo(const Foo&)'
    3 |     Foo(Foo const&) {}
      |     ^~~
```
for `gcc` and
```
<source>:3:5: error: definition of implicit copy assignment operator for 'Foo' is deprecated because it has a user-provided copy constructor [-Werror,-Wdeprecated-copy-with-user-provided-copy]
    Foo(Foo const&) {}
    ^
<source>:11:10: note: in implicit copy assignment operator for 'Foo' first required here
    foo1 = foo2;
         ^
```
for `clang`.

Reply:
Very good.  Thanks.